# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fetch/FetchStreamUtils.cpp
# Commit: 4fd92351d64b
# Full Hash: 4fd92351d64b1cbeb82beaf6228b21c01dc9bc03
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-01-19 04:46:07
# Regressor Bug: 1351231
# File Overlap Count: 1
# Description:
#   Bug 1351231 - FetchService integration for PFetch. r=dom-worker-reviewers,jesup
#   
#   Depends on D142436
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D142437
# ==============================================================================

diff -r 7a4e3f5f674a -r 4fd92351d64b dom/fetch/FetchStreamUtils.cpp
--- a/dom/fetch/FetchStreamUtils.cpp	Wed Jan 18 00:50:19 2023 +0000
+++ b/dom/fetch/FetchStreamUtils.cpp	Wed Jan 18 00:50:20 2023 +0000
@@ -9,6 +9,7 @@
 #include "mozilla/RemoteLazyInputStreamStorage.h"
 #include "mozilla/dom/FetchTypes.h"
 #include "mozilla/dom/IPCBlob.h"
+#include "mozilla/ipc/IPCStreamUtils.h"
 #include "nsContentUtils.h"
 #include "nsXULAppAPI.h"
 
@@ -34,7 +35,12 @@
 NotNull<nsCOMPtr<nsIInputStream>> ToInputStream(
     const ParentToChildStream& aStream) {
   MOZ_ASSERT(XRE_IsContentProcess());
-  nsCOMPtr<nsIInputStream> result = aStream.stream();
+  nsCOMPtr<nsIInputStream> result;
+  if (aStream.type() == ParentToChildStream::TRemoteLazyInputStream) {
+    result = aStream.get_RemoteLazyInputStream();
+  } else {
+    result = DeserializeIPCStream(aStream.get_IPCStream());
+  }
   return WrapNotNull(result);
 }
 
@@ -50,11 +56,20 @@
 
 ParentToChildStream ToParentToChildStream(
     const NotNull<nsCOMPtr<nsIInputStream>>& aStream, int64_t aStreamSize,
-    NotNull<mozilla::ipc::PBackgroundParent*> aBackgroundParent) {
+    NotNull<mozilla::ipc::PBackgroundParent*> aBackgroundParent,
+    bool aSerializeAsLazy) {
   MOZ_ASSERT(XRE_IsParentProcess());
 
   ParentToChildStream result;
-  result.stream() = RemoteLazyInputStream::WrapStream(aStream.get());
+  if (aSerializeAsLazy) {
+    result = RemoteLazyInputStream::WrapStream(aStream.get());
+  } else {
+    nsCOMPtr<nsIInputStream> stream(aStream.get());
+    mozilla::ipc::IPCStream ipcStream;
+    Unused << NS_WARN_IF(
+        !mozilla::ipc::SerializeIPCStream(stream.forget(), ipcStream, false));
+    result = ipcStream;
+  }
   return result;
 }
 